src/Sylius/Bundle/UserBundle/Doctrine/ORM/UserRepository.php

```diff
@@ -155,4 +155,23 @@ public function findOneByEmail($email)
             ->getOneOrNullResult()
         ;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function findOneByToken($token)
+    {
+        $queryBuilder = $this->createQueryBuilder('o');
+
+        $queryBuilder
+            ->orWhere($queryBuilder->expr()->eq('o.passwordResetToken', ':token'))
+            ->orWhere($queryBuilder->expr()->eq('o.emailVerificationToken', ':token'))
```

Comment:
That is wrong. This way we can use verification token to reset password and the other way around. We should explicitly look for each token separately.